ui/apps/dashboard/src/queries/restAPIHooks.ts

```diff
@@ -0,0 +1,42 @@
+'use client';
+
+import { useEffect, useState } from 'react';
+import { useAuth } from '@clerk/nextjs';
+
+export function useRestAPIRequest<T>({
+  url,
+  method,
+}: {
+  url: string | URL | null;
+  method: string;
+}): { data: T | undefined; fetching: boolean; error: Error | null } {
```

Comment:
Could be nice to share the `FetchResult` type with `useGraphQLQuery`. That type is a discriminated union, which makes consumption nicer: if you handle the loading and error states then you're left with non-nullish `data` property

https://github.com/inngest/inngest/blob/main/ui/apps/dashboard/src/utils/useGraphQLQuery.ts#L39